src/tess_asteroids/utils.py

```diff
@@ -95,3 +98,247 @@ def compute_moments(flux, mask=None):
+        ax.scatter(xy[1], xy[0], marker="o", c="red", alpha=1, s=15)
+
+    ax.set_aspect("equal", "box")
+    ax.set_title(title)
+
+    if aperture_mask is not None:
+        row, col = np.mgrid[
+            corner[0] : corner[0] + img.shape[0], corner[1] : corner[1] + img.shape[1]
+        ]
+        for i, pi in enumerate(row[:, 0]):
+            for j, pj in enumerate(col[0, :]):
+                if aperture_mask[i, j]:
+                    # print("here")
+                    rect = patches.Rectangle(
+                        xy=(pj - 0.5, pi - 0.5),
+                        width=1,
+                        height=1,
+                        color="tab:red",
+                        fill=False,
+                        # hatch="//",
+                        alpha=0.8,
+                    )
+                    ax.add_patch(rect)
+
+    ax.set_xlabel("Pixel Column")
+    ax.set_ylabel("Pixel Row")
+
+    return ax
+
+
+def animate_cube(
+    cube,
+    aperture_mask=None,
+    corner=[0, 0],
+    track=None,
+    cadenceno=None,
+    time=None,
+    interval=200,
+    repeat_delay=1000,
+    cnorm=False,
+    suptitle="",
+):
+    """
+    Creates an animated visualization of a 3D image cube, with an optional aperture mask and other customization options.
+
+    This function animates the slices of a 3D image cube, optionally overlaying an aperture mask,
+    and provides controls for animation speed, title, and tracking information.
+
+    Parameters:
+    -----------
+    cube : 3D array
+        A 3D array representing the image cube (e.g., a stack of 2D images over time).
+
+    aperture_mask : 2D or 3D array, optional
+        A binary mask (same shape or a 2D slice of `cube`) to overlay on each frame of the animation.
+        If a 2D mask is passed, it will be repeated for all times.
+
+    corner : list of two ints, optional, default=[0, 0]
+        The (row, column) coordinates of the lower left corner of the image.
+
+    track : list or tuple, optional, default=None
+        A list or tuple of object positions to be displayed on the plot.
+        For proper display of object position, if corner is [0, 0] then track needs to be relative to corner.
+        If corner is provided, track needs to be absolute.
+        If None, no tracking information is shown.
+
+    cadenceno : int, optional, default=None
+        The cadence number of the frames, used for information display.
+
+    time : array-like, optional, default=None
+        Array of time values corresponding to the slices in the cube.
+
+    interval : int, optional, default=200
+        The time interval (in milliseconds) between each frame of the animation.
+
+    repeat_delay : int, optional, default=1000
+        The time delay (in milliseconds) before the animation restarts once it finishes.
+
+    cnorm : optional, default=False
+        Weather to use asinh color normalization (from astropy.visualization.simple_norm).
+
+    suptitle : str, optional, default=""
```

Comment:
Would it be better to use default suptitle = None, or does that not work with formatting of figure?